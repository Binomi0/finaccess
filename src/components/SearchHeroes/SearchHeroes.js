import React, { useState } from 'react';
import {
  SearchHeroesContainer,
  InputHeroes,
  SearchButton,
} from '../styles/searchHeroes';

const SearchHeroes = ({ handleChange }) => {
  const [name, setName] = useState('Thor');

  return (
    <SearchHeroesContainer>
      <InputHeroes
        name="heroe"
        type="text"
        value={name}
        onChange={({ target }) => setName(target.value)}
      />
      <SearchButton onClick={() => handleChange(name)}>
        <span role="img" aria-label="magnifier">
          🔍
        </span>
      </SearchButton>
    </SearchHeroesContainer>
  );
};

export default SearchHeroes;
