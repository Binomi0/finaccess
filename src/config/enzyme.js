import { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

exports.mount = mount;
exports.shallow = shallow;
