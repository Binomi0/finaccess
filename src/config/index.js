const random = Math.floor(Math.random() * 1000);
const results = 3;

export const apikey = '2b338eb5f577d71ecf79c2347a55dc75';
export const filters = `&limit=${results}&offset=${random}`;
