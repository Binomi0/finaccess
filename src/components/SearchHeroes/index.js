import SearchHeroes from './SearchHeroes';

export default SearchHeroes;
