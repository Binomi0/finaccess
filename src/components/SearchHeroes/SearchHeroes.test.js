import React from 'react';
import { mount } from '../../config/enzyme';
import SearchHeroes from './SearchHeroes';

describe('SearchHeroes', () => {
  it('should renders without crashing', () => {
    const handleChange = jest.fn();
    const Search = mount(<SearchHeroes handleChange={handleChange} />);

    Search.find('button').simulate('click');
    Search.find('input').simulate('change');

    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(Search.html()).toMatchSnapshot();
  });
});
