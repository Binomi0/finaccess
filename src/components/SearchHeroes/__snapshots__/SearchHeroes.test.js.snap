// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeroes should renders without crashing 1`] = `"<div class=\\"sc-bdVaJa iHYTGn\\"><input name=\\"heroe\\" type=\\"text\\" class=\\"sc-bwzfXH dErYyH\\" value=\\"Thor\\"><button class=\\"sc-htpNat icYeav\\"><span role=\\"img\\" aria-label=\\"magnifier\\">🔍</span></button></div>"`;
